2016/exercises/final_project/Alba-Petar-Rebecca-Simon/PolytopeProject/Gale_Diagram.cpp

```diff
@@ -0,0 +1,322 @@
+//
+//  Gale_Diagram.cpp
+//  PolytopeProject
+//
+
+
+#include "Gale_Diagram.hpp"
+
+
+
+Gale_Diagram::Gale_Diagram(vector<Point> points): points(points),matrix(Matrix(points)){
+};
+
+bool Gale_Diagram::is_neighborly() const{
+    //Choose 2 points
+    for(int i=0; i<points.size()-1; i++){
+        for(int j=i+1; j<points.size(); j++){
+            
+            bool edgeIJPresent=false;
+            
+            
+            //Then choose another 2
+            for(int k=0; k<points.size()-1 && (!edgeIJPresent); k++){
+                for(int l=k+1; l<points.size() && (!edgeIJPresent); l++){
+					if( (i!=k) and (i!=l) and (j!=k) and (j!=l)){
+						edgeIJPresent = isSimplex(i, j, k, l);
+					}
+                }
+            }
+            
+            if(!edgeIJPresent){
+                return false;
+            }
+        }
+        
+    }
+    return true;
+}
+
+
+
+vector<int> Gale_Diagram::findMissing(int i, int j, int k, int l) const{
+    vector<int> pointNums(points.size());
+    for(int a=0; a<points.size(); a++) pointNums[a]=a;
+
+	pointNums[i] = -1;
+	pointNums[j] = -1;
+	pointNums[k] = -1;
+	pointNums[l] = -1;
+	
+    vector<int> pN;
+    
+    for(int a=0; a<points.size(); a++){
+		if(pointNums[a]>=0) pN.push_back(a);
+	}
+    
+    return pN;
+}
+
```

Comment:
Also, the name is a little misleading because the function actually returns the nonnegative indices, but i,j,k,l I guess are supposed to be the missing ones